### DISCUSSION TITLE
How do I program GO Midi Send for Stops?

### DISCUSSION THREAD

**Discussion by larspalo** [GrandOrgue/grandorgue#361] [General]
Originally posted 2020-12-04 15:21:02.949000 by djsorg2062 in GrandOrgue discussion forum on SourceForge<br><br>How can I program the GO Midi-Settings for Stops to Send by Copying the current receive device? I program the Receive ok and attempt to copy to Send, but there is no response in Send. Do I have to manually enter the send data for each stop? I use Mojave in a MacBook Pro.

Also, when Listening for Event, the default Upper Limit is 1, should I manually set it to 127?<br><br>Attachments in original post:<br>https://sourceforge.net/p/ourorgan/discussion/962124/thread/9db2763f31/e129/attachment/22415F73-0A04-4223-A67B-EF78CC66CDC2.jpeg<br>https://sourceforge.net/p/ourorgan/discussion/962124/thread/9db2763f31/e129/attachment/366E262F-2DAB-443F-BBD6-CF65774F0A4A.jpeg<br>

**Comment by larspalo** [comment]
Originally posted 2020-12-04 15:28:54.429000 by [csw900](https://github.com/midimusic) in GrandOrgue discussion forum on SourceForge<br><br>The upper limit for keyboard note events should be set to 127. Do it by hand if it does not happen automatically. The lower limit should always be 1.

I also have tried to copy stop events to the output. As far as I can tell it does not work. I always do it manually. If anyone has got this working please respond and tell us how.

csw900

**Comment by ahall41** [comment]
The SEND events seem to bear no resemblance to RECV. My Johannus 170 sends a Cx Program Change, but I can only choose Cx Program Change On or Cx Program Change Of for output - neither of which appears to work (the listed parameters look different too). I've had SEND events working OK with other software. 

I expect this is going to be quite a bit of work to sort out. GO doesn't seem to handle SysEx events particularly well, for example (I tried fiddling with them to get my SubZero ControlPad working).

**Reply by wpostma** [reply]
I am curious if it is possible to get the Novation Launchpad to function as an organ stop controller.  If anyone knows if custom scripts could be written for Grande Orgue to permit complex midi send out logic to function, it would be great.
The way the launchpad works is there is an 8x8 primary grid of notes, and you would be writing back to it an on off status (led light color) to indicate the current stop on/off state.
